### PR TITLE
Random Names Rework Part 2 - MekHQ Changes

### PR DESCRIPTION
**Pull request by Windchild292**
This is part two of my rework of the random name system.

The MekHQ change is that the origin faction is now used to generate instead of the player's faction, as that provides a better format for generating names.

This looks far worse a PR than it is because of the data file changes. Most of the files that are changed are just a few lines of changes, but gender and the generators do touch on a bunch of files. The data files MUST be updated immediately or MekHQ will not be able to generate names.

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/mission/AtBScenario.java [hunk 1]
```diff
@@ -1660,7 +1660,7 @@ public static String writeEntityWithCrewToXmlString(Entity tgtEnt, int indentLvl
             crew.append(MekHqXmlUtil.indentStr(indentLvl + 2)).append("<crewMember slot=\"")
                     .append(pos).append("\" name=\"").append(MekHqXmlUtil.escape(tgtEnt.getCrew().getName(pos)))
                     .append("\" nick=\"").append(MekHqXmlUtil.escape(tgtEnt.getCrew().getNickname(pos)))
-                    .append("\" gender=\"").append(tgtEnt.getCrew().getGender(pos))
+                    .append("\" gender=\"").append(tgtEnt.getCrew().getGender(pos).name())
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/mission/AtBScenario.java:1663]: Is this the code that writes out a MUL entry? Can the MUL Parser handle this change?

File: MekHQ/src/mekhq/campaign/personnel/Person.java [hunk 2]
```diff
@@ -2103,7 +2097,7 @@ public static Person generateInstanceFromXML(Node wn, Campaign c, Version versio
                 } else if (wn2.getNodeName().equalsIgnoreCase("hits")) {
                     retVal.hits = Integer.parseInt(wn2.getTextContent());
                 } else if (wn2.getNodeName().equalsIgnoreCase("gender")) {
-                    retVal.gender = Integer.parseInt(wn2.getTextContent());
+                    retVal.gender = Gender.parseFromString(wn2.getTextContent().trim());
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/personnel/Person.java:2100]: Does parseFromString handle gender entries from older campaigns?

Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/personnel/Person.java:2100]: It does. It defaults to valueOf but does enable integer parsing.

File: MekHQ/src/mekhq/campaign/mission/AtBScenario.java (hunk 1 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/mission/AtBScenario.java:1663]: It can, using the same Gender::parseFromString method as Person does to load.

**Pull request merged**